### PR TITLE
Jon up keep

### PR DESCRIPTION
**Pull request by JonathanStefanov**
Set of functions that implement [ChainLink Keepers](https://chain.link/keepers) in order to regularly check if the project is funded or not

### CODE REVIEW

File: contracts/EtherCrowd.sol [hunk 1]
```diff
@@ -114,7 +119,56 @@ contract EtherCrowd {
...
+            }
+        }
+    }
+
+    function endCrowdsale(Crowdsale memory _crowdsale) private {
+        // TODO: Emit events
+        // Crowdsale has ended, refund the money or give it to the owner according to the situation
+        if(_crowdsale.currentAmount >= _crowdsale.goalAmount){
+            // The crowdsale is successful, we can transfer the money to the owner
+            payable(_crowdsale.owner).transfer(_crowdsale.currentAmount);
```
Comment (ghost) [inline contracts/EtherCrowd.sol:150]: Should we make this a function to improve readability ?

File: contracts/EtherCrowd.sol [hunk 2]
```diff
@@ -114,7 +119,56 @@ contract EtherCrowd {
...
+        else{
+            // refunding the investors
+            refund(_crowdsale);
+        }
+
+    }
+
+    function refund(Crowdsale memory _crowdsale) private {
+            for (uint i=0; i< _crowdsale.contributors.length ; i++) {
+                address contributor = _crowdsale.contributors[i];
```
Comment (ghost) [inline contracts/EtherCrowd.sol]: Rename to contributorAddress to improve readability.

File: contracts/EtherCrowd.sol [hunk 3]
```diff
@@ -57,11 +62,11 @@ contract EtherCrowd {
         string memory _videoUrl,
         string memory _thumbnailUrl,
         string memory _description,
-        uint _startDate,
+        //  no need to specify start date, it is the time at which the user calls the function
         uint _endDate
 
     ) external payable{
-        require(_endDate > _startDate, "End date has to be after start date");
```
Comment (ghost) [inline contracts/EtherCrowd.sol:57]: We must implement startDate, it is an essential feature.

**Review by ghost — COMMENTED** [review]
A few things to change but overall it's good.

File: contracts/EtherCrowd.sol (hunk 1 above)
Comment (JonathanStefanov) [reply contracts/EtherCrowd.sol:150]: A function just for one line? Not sure it will improve much tbh

File: contracts/EtherCrowd.sol (hunk 2 above)
Comment (JonathanStefanov) [reply contracts/EtherCrowd.sol]: Will do!

File: contracts/EtherCrowd.sol (hunk 3 above)
Comment (JonathanStefanov) [reply contracts/EtherCrowd.sol:57]: This line has been removed since last commit

**Pull request merged**